### DISCUSSION TITLE
Cloudflare Zero Trust validation

### DISCUSSION THREAD

**Discussion by kekonn** [authelia/authelia#9667] [Ideas]
### Description

Use the cookie/JWT token Cloudflare Zero Trust sets on every request it tunnels, as an authentication method. This is an easier method than the OIDC 1 version currently supported.

### Use Case

When employees are logged onto Cloudflare Zero Trust, this cookie is set for every subdomain. Thus if Authelia reads and validates the cookie, we know the user is authenticated. You can then match the email from the OIDC claims in the JWT to Authelia's user database. If the user is present, simply consider him logged in.

### Details

It is fairly trivial to [validate this cookie](https://developers.cloudflare.com/cloudflare-one/identity/authorization-cookie/validating-json#programmatic-verification) with cloudflare, provided you already have a way to validate JWT tokens (I assume you do).

The only setting needed would be an API key that the user creates. The JWT token always contains the user's email, which would be everything that Authelia needs to do it's thing.

It is also a lot easier to set up than an OIDC flow, since the presence of a valid cookie means Cloudflare already did the OIDC checks.

### Documentation

https://developers.cloudflare.com/cloudflare-one/identity/authorization-cookie/validating-json#programmatic-verification

https://developers.cloudflare.com/api/

### Pre-Submission Checklist

- [X] I agree to follow the [Code of Conduct](http://www.authelia.com/code-of-conduct)

- [X] I have checked for related issues and checked the documentation

**Comment by james-d-elliott** [comment]
I've thought about this a few times. They don't clearly communicate their API in my opinion. If they communicated via the documentation the claims they included from the OIDC backend and how this was accomplished we could decide if this was at all possible.

Realistically they should return at the very least the `sub`, `iss`, and `amr` claims of the OIDC backend. Then we can just check those against the Authelia backend for who the user is.

**Comment by kekonn** [comment]
I'll see if I can find some time to check which claims they include

**Reply by huyz** [reply]
Any progress since?
